yarpl/include/yarpl/flowable/Subscriber.h

```diff
@@ -2,31 +2,41 @@
 
 #pragma once
 
-#include <folly/ExceptionWrapper.h>
-#include <stdexcept>
-
 #include "yarpl/Refcounted.h"
 #include "yarpl/flowable/Subscription.h"
 
+#include <folly/ExceptionWrapper.h>
+
+#include <glog/logging.h>
+
 namespace yarpl {
 namespace flowable {
 
 template <typename T>
 class Subscriber : public virtual Refcounted {
  public:
-  // Note: if any of the following methods is overridden in a subclass,
-  // the new methods SHOULD ensure that these are invoked as well.
+  // Note: If any of the following methods is overridden in a subclass, the new
+  // methods SHOULD ensure that these are invoked as well.
   virtual void onSubscribe(Reference<Subscription> subscription) {
-    subscription_ = subscription;
+    DCHECK(subscription);
+
+    if (subscription_) {
+      subscription->cancel();
```

Comment:
We probably want to complain loudly that we called onSubscribe twice

Reply:
I believe this actually fine, as there are some operators that re-subscribe (Rx.repeat I think), and that's okay.